### ISSUE TITLE
Weighted Log Likelihood #featurerequest

### ISSUE THREAD

**Issue by deveshjawla** [TuringLang/AdvancedHMC.jl#317]
For imbalanced datasets a weighted loss function works better than either oversampling or under sampling. Can we add this feature in Turing.jl and AdvancedHMC's NUTS? The log likelihood function would require a weight term added for each class, and according to the number of samples in the training dataset, we would scale the corresponding log likelihood term by the inverse of the number of the samples for that class. I would be happy to contribute but can't find where is it located in the source code, could someone please help?

**Comment by deveshjawla** [comment]
Closing because Stale. Workaround as follows:

If `weights_vector` is the list of weights of each sample in train target vector `y`, then inside a Turing probabilistic model, one can do as follows (below a case of classification):
```
for i ∈ 1:lastindex(y)
		loglik = loglikelihood(Categorical(softmax(preds[:, i])), y[i]) * weights_vector[i]
		Turing.@addlogprob!(loglik)
	end
```
